agendrix/deepface
# Customize Deepface for Agendrix usage

# General changes
- Setup PDM
- Add Agendrix-specific README
- Add benchmarking script

# Deepface modifications
- Allow passing multiple faces embeddings to `verify` instead of only 1 face.

# Agendrix-specific modifications
- Add scripts and utils for face detection + verification
- Add Redis connection for posting results

# How to test
- You can use the data in `tests/agendrix-dataset/stagin-data` for simple tests
- Otherwise, I use the [Labeled Faces in the Wild](https://vis-www.cs.umass.edu/lfw/#download) dataset ([direct download link](http://vis-www.cs.umass.edu/lfw/lfw.tgz)).

```
pdm detect IMG_PATH
```

```
pdm verify IMG1_PATH IMG2_PATH
```

```
pdm run tests/benchmark_models.py ~/Downloads/faces/lfw/ --config OpenFace,mediapipe --config Facenet512,mediapipe --config OpenFace,yolov8 --config VGG-Face,opencv --n-samples 5 --output benchmark.csv
```
